src/nl/hannahsten/texifyidea/refactoring/inlinefile/LatexInlineFileProcessor.kt

```diff
@@ -0,0 +1,87 @@
+package nl.hannahsten.texifyidea.refactoring.inlinefile
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.command.CommandProcessor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiReference
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.search.SearchScope
+import com.intellij.psi.search.searches.ReferencesSearch
+import com.intellij.refactoring.BaseRefactoringProcessor
+import com.intellij.usageView.UsageInfo
+import com.intellij.usageView.UsageViewDescriptor
+import nl.hannahsten.texifyidea.file.LatexFile
+
+/**
+ * Recieves settings and a target and performs the requested refactoring
+ *
+ * @see com.intellij.refactoring.inline.InlineMethodProcessor
+ *
+ * @author jojo2357
+ */
+class LatexInlineFileProcessor(
+    myProject: Project,
+    private val inlineFile: LatexFile,
+    private val originalReference: PsiElement?,
+    private val isInlineThisOnly: Boolean,
+    private val isKeepTheDeclaration: Boolean,
+    private val myScope: SearchScope = GlobalSearchScope.projectScope(myProject)
+) : BaseRefactoringProcessor(myProject) {
+
+    override fun createUsageViewDescriptor(usages: Array<out UsageInfo>): UsageViewDescriptor {
+        return LatexInlineFileDescriptor(inlineFile)
+    }
+
+    override fun findUsages(): Array<UsageInfo> {
+        val tempreferences = ReferencesSearch.search(inlineFile, myScope).findAll().asSequence()
+
+        return tempreferences
+            .distinct()
+            .map(PsiReference::getElement)
+            .map(::UsageInfo)
+            .toList()
+            .toTypedArray()
+    }
+
+    override fun performRefactoring(usages: Array<out UsageInfo>) {
+        if (isInlineThisOnly) {
+            if (originalReference != null)
```

Comment:
I think it can be null if intellij is still resolving the file? In that case just do nothing I'd say

Reply:
It seems like everything is broken actually

Reply:
Oh wait I misunderstood, it is throwing but still works...

Reply:
CRAP I lost the `else` branch. Fixing now

Reply:
Please try to replicate the error without the bug. I still think an error is appropriate since it signifies an internal error has occurred.